libnd4j/include/ops/declarable/platform/armcompute/armcomputeUtils.cpp

```diff
@@ -0,0 +1,291 @@
+namespace sd      {
+namespace ops       {
+namespace platforms {
+
+
+
+Arm_DataType getArmType ( const DataType &dType){
+     Arm_DataType  ret;
+     switch (dType){  
+        case HALF :
+            ret = Arm_DataType::F16;
+            break;        
+        case FLOAT32 :
+            ret = Arm_DataType::F32;
+            break;
+        case DOUBLE :
+            ret = Arm_DataType::F64;
+            break;
+        case INT8 :
+            ret = Arm_DataType::S8;
+            break;
+        case INT16 :
+            ret = Arm_DataType::S16;
+            break;
+        case INT32 :
+            ret = Arm_DataType::S32;
+            break;
+        case INT64 :
+            ret = Arm_DataType::S64;
+            break;
+        case UINT8 :
+            ret = Arm_DataType::U8;
+            break;
+        case UINT16 :
+            ret = Arm_DataType::U16;
+            break;        
+        case UINT32 :
+            ret = Arm_DataType::U32;
+            break;        
+        case UINT64 :
+            ret = Arm_DataType::U64;
+            break; 
+        case BFLOAT16 : 
+            ret = Arm_DataType::BFLOAT16;
+            break;
+        default:
+            ret = Arm_DataType::UNKNOWN;
+     }; 
+
+    return ret;
+}
+bool isArmcomputeFriendly(const NDArray& arr) {
+  auto dType = getArmType(arr.dataType());
+  int rank = (int)(arr.rankOf());
+  auto arrStrides = arr.stridesOf();
+  return dType != Arm_DataType::UNKNOWN && arr.ordering() == 'c' &&
+         arrStrides[rank - 1] == 1;
+}
+
+Arm_TensorInfo getArmTensorInfo(int rank, Nd4jLong* bases,sd::DataType ndArrayType, arm_compute::DataLayout layout) {
+    constexpr int numChannels = 1; 
+    auto dType = getArmType(ndArrayType);
+    if (dType == Arm_DataType::UNKNOWN) {
+        //should not happen when isArmcomputeFriendly check is used
+        throw std::runtime_error("unsupported Data Type");
+    }
+
+    if (rank > arm_compute::MAX_DIMS) {
+        throw std::runtime_error("the rank of the array is higher");
+    } 
+    Arm_TensorShape shape;
+    shape.set_num_dimensions(rank); 
+    for (int i = 0, j = rank - 1; i < rank; i++, j--) {
+        shape[i] = static_cast<uint32_t>(bases[j]); 
+    }
+    // fill the rest unused with 1
+    for (int i = rank; i < arm_compute::MAX_DIMS; i++) {
+        shape[i] = 1;
+    } 
+
+    return Arm_TensorInfo(shape, numChannels, dType, layout); 
+}
+
+Arm_TensorInfo getArmTensorInfo(const NDArray& arr,
+                                arm_compute::DataLayout layout) {
+  auto dType = getArmType(arr.dataType());
+  if (dType == Arm_DataType::UNKNOWN) {
+    throw std::runtime_error("unsupported Data Type");
+  }
+
+  //
+  constexpr int numChannels = 1;
+  int rank = (int)(arr.rankOf());
+  auto bases = arr.shapeOf();
+  auto arrStrides = arr.stridesOf();
+  if (rank > arm_compute::MAX_DIMS) {
+    throw std::runtime_error("the rank of the array is higher");
+  }
+  if (arr.ordering() != 'c' || arrStrides[rank - 1] != 1) {
```

Comment:
Same as above

Reply:
well. tbh my conclusion of the last stride being 1 could be faulty. but I believe even for NHWC it should be so.